### PR TITLE
Adds fancy boxes to cardboard crafting & flour sacks + UE designs to biogenerator.

### PR DESCRIPTION
**Pull request by FantasticFwoosh**
:cl: Moon Farmer
add: Both reagent universal enzyme & sacks of flour are now availible for their respective costs from the biogenerator.
add: New crafting recipies for donut boxes, eggboxes & candle boxes have been added to cardboard recipies for the collective benefit of service personnel and the station.
/:cl:

thanks to idea contributors

This PR strives to achieve greater co-operation by a use of biogenerator supplies, firstly the chef directly benefits with added capacity to store more donuts :doughnut: & additional eggs :egg: when availible than their starting equipment from constructable fancy boxes, while also being able to quickly produce quick sacks of wheat & spare universal enzyme with only minor botany input instead of waiting for lengthy emergency imports from cargo.

- The cost of the wheat sacks incentivises botanists who are too busy to grow wheat directly to quickly supply the chef out of other chaff plants, while making it ineffective enough to not rely upon it long-term.

:atom_symbol: **Why couldn't you atomise this? :** In practicality outside the scope of the game it is being conservative of GBP. Also, many of the points of fancy objects revolve around food and their relationship between service, such as the eggs to make the pastry, which then in turn makes the doughnuts to be eaten at a candlelit dinner (_if so inclined_) neatly stored in quantity. :candle:

### CODE REVIEW

**Comment by FantasticFwoosh** [comment]
**Image related**: back from my testing, everything works fine in play as i wanted it to as i tried each individual aspect, everything you can make is on show on the wooden tables & kitchen counter (except maybe the wine bottle and candles themselves)
captain on a date in a up-dog restaurant

**Comment by subject217** [comment]
What's the point of the spare universal enzyme? It's a catalyst, so it's never consumed. You only ever need 1u of it.

**Comment by FantasticFwoosh** [comment]
Its still easy to be careless with it or lose it, which hence is why the food crate from cargo previously was the only way to get more. The x5 max dispense rate of the biogenerator quite neatly transfers into a brand new condiment bottle when ran through the condiment-master machine for the cost of 150 biomass. 

A biogenerator seperately can also be used to kickstart a entire kitchen when additionally supplied with ingredients from plants, which is partially intentional for those without access for UE at all.

**Comment by BebeYoshi** [comment]
You print sack of flour... full of flour? Even though you can easily plant some wheat and just fill the sack with it? I can see why did you make it but why do we need it, wheat has by default production speed 1, barely taking a minute to grow, it also has yield 5 and you can easily put potency 50 and nutriment 20% for a lot of flour, filling sacks of flour (which makes more sense them printing them full already). Also as said before, cooks won't waste universal enzyme at all so it is kind of pointless as well. I may have misunderstood what you meant so if I'm wrong you are free to ignore all of this.

**Comment by Tlaltecuhtli** [comment]
add welding goggles to leather craft menu as they are unobtainable and important for moth people

**Comment by FantasticFwoosh** [comment]
BebeYoshi i put my core reasoning in the OP; inefficient quick hack for people too busy to wait, of course i endorse growing wheat by yourself as a more optimal high yield method and the sacks are not meant to be replacements just temporary suppliments.

> The same way you could milk pete or deliver a cow from cargo yourself, but you also have the option to bio-generator a full milk carton from the biogen.

Its more convenient and also consistent for UE to be generated this way along with other reagents like milk, cream & black pepper (biogen exclusive) by comparison other than rely on it to arrive by cargo food-crate delivery, keeping in mind with this change that you can use it for other things (flavours etc) without having to worry about spending it too easily.

**Comment by BebeYoshi** [comment]
... Except you can't get milk by planting, that's why it's on the biogenerator, which is meant to be a machine that can makes things you can't get from plants, the same time you will wait to use the biogenerator and get sack of flour you can use to plant wheat.

**Comment by Barhandar** [comment]
Does amount of flour you get per wheat scale with potency/amount of nutriment? Because biogenerator output does scale with amount of nutriment and therefore it just might be better to plant pumpkins and feed them into generator, rather than wheat.

**Comment by BebeYoshi** [comment]
You can insert 20% nutriment in any plant. So 50 potency wheat with 20% nutriment should be able to supply the kitchen and the biogenerator for the entire shift in less than 7 minutes. Even faster if you plant it in 2 trays.

**Comment by FantasticFwoosh** [comment]
Im not going to argue with you, soymilk is growth exclusive from soybeans, you can feed farm animals wheat for more produce. You can grow a variety of other plants this way without priotitizing trays for wheat, what you want to do is your own personal botany playstyle is supported regardless. This complaint _**feels**_ trivial not to say that it is.

**Comment by BebeYoshi** [comment]
Soymilks and Space Milk have different uses though, that's why you can get milk from the biogenerator, not soymilk

**Review by Tlaltecuhtli — APPROVED** [review]
it compiles so merge this

File: code/modules/research/designs/biogenerator_designs.dm [hunk 1]
```diff
@@ -51,6 +51,22 @@
...
+	materials = list(MAT_BIOMASS = 30)
+	make_reagents = list("enzyme" = 10)
+	category = list("initial","Food")
+
+/datum/design/flour_sack
+	name = "Flour Sack"
+	id = "flour_sack"
+	build_type = BIOGENERATOR
+	materials = list(MAT_BIOMASS = 150)
+	build_path = /obj/item/reagent_containers/food/condiment/flour
```
Comment (Iamgoofball) [inline code/modules/research/designs/biogenerator_designs.dm]: what

just grow wheat smh seriously how high are you

im reverting this

**Comment by octareenroon91** [comment]
Put `enzyme` in `pineapples`.

**Comment by FantasticFwoosh** [comment]
I was going to but i didn't PR pineapples in the end, i just supplied the growth sprites out of what i had of the project. I can't remember if i was gitbanned at the time or not. Biogenerator is a fine alternative to adding pepper plants.

**Comment by ExcessiveUseOfCobblestone** [comment]
tfw powergaming botanists can't be arsed with making wheat so they have to put their mega healing/death plants in the biogen for flour.

If it makes flour it should be poor quality and the food should taste bad as a result, otherwise it should be removed.

**Comment by ExcessiveUseOfCobblestone** [comment]
Wheat is not the best plant t. Mimana Holmes (best botanist NA)

**Comment by Bawhoppen** [comment]
Please remove the flour change.

**Comment by FantasticFwoosh** [comment]
> 
> 
> Please remove the flour change.

I appreciate you not making as much a fuss as other posters and asking directly & politely, but initially I think the backlash will die away once its merged into stratum of how the chef & botany interact. Goof's already entrusted their assurance eitherway but i don't feel that there is a need to make such a change.

**Comment by ExcessiveUseOfCobblestone** [comment]
This is why bundling feature additions is cancer.

**Comment by 81Denton** [comment]
The only people who will benefit from flour printing are chefs that make food, i.e. the ones that actually do their job and end up making the round more enjoyable for other players

I get the "fuk botany" sentiment, but this is honestly the wrong change to bikeshed about

**Comment by BebeYoshi** [comment]
It will also "buff" perma since you can now easily make dough for real food

**Comment by ExcessiveUseOfCobblestone** [comment]
@81Denton if they have access to a biogen then certainly they can move hydroponics into their kitchen?

this is certainly the hill i'd like to mildly irritate my skin on if we're trying to streamline a process because people aren't doing the job properly.

**Comment by octareenroon91** [comment]
>It will also "buff" perma since you can now easily make dough for real food

Imagine stripping designs from the perma biogenerator.

**Comment by 81Denton** [comment]
@ExcessiveUseOfCobblestone Pubbystation has the biogenerator as public access, which IMO is a good thing since it lets botanists contribute more. I should probably add that to other maps as well.

**Comment by FantasticFwoosh** [comment]
Because it dispenses cardboard & a number of other things also, putting a biogenerator on the mining base or cargo would allow miners to use the plant bags even if the aren't delivering produce to the station.

- Plant bags fit inside explorer belts fine and intentionally for space conservation since weight is ignored in regards to huge objects like sonic hammers fitting in fine, and can mass collect plant ash flora for usually high returns.

**Comment by ExcessiveUseOfCobblestone** [comment]
I wouldn't mind biogen public access, all chefs should have by default hydroponics access tbh.

**Comment by ExcessiveUseOfCobblestone** [comment]
You're welcome to search PRs that put the same focus that same relationship between MDs and Chemists as Chef and Hydroponics, and you realize every time they do at least one person says the MD becomes useless. An example is 32929.

Chef becomes useless without hydroponic's assistance, I agree entirely with that. The fine thread here is that I don't think they should be able to forgo actually doing the job to get the rewards. 

It would be on par with not being able to function properly without chem as doc, so you go to chem and you have a chem converter that converts all the naughty poisons, lube, etc. into medicine. Doesn't seem to make much sense.

**Comment by BebeYoshi** [comment]
I think this is intended and is a good thing, not only that but I think all departaments should be like that (depending one another, so it's not just a singleplayer thing where you only depends on your actions), even if there should probably be a disadvantage if botany doesn't plants for cook or the same for chem and MD.

If cook had a access to botany would just split between plant researchers and people that plant for food (which in my personal opnion is a very bad thing).

Also that's what heads are for, demote people that are intentionally not doing their job or giving access if there is no botanist etc.

**Comment by ExcessiveUseOfCobblestone** [comment]
But the feature here allows you to bypass that using anything the botanist makes, so we agree lol.

**Comment by BebeYoshi** [comment]
I don't think adding flour sack would make botany or kitchen any better, if the botanists are willing to use the biogenerator for you, they could easily get way better results by planting wheat/tomatoes/whatever, I don't think this PR is bad, neither public biogenerator (not related to this PR), I think that having flour sacks in them are a really bad ideia and cooks having acess to botany(also not related to this PR) but since the author already said they won't change that anything I or you say here won't change their mind

**Comment by vuonojenmustaturska** [comment]
eh fuck let's try it, this PR really didn't need 40 comments worth of debate about it

**Pull request merged**